cmd/cvc-operator/controller/cvc_controller.go

```diff
@@ -258,7 +258,7 @@ func (c *CVCController) createVolumeOperation(cvc *apis.CStorVolumeClaim) (*apis
 	}
 
 	klog.V(2).Infof("creating cstorvolume replica resource")
-	err = c.distributePendingCVRs(cvc, cvObj, svcObj)
+	err = c.distributePendingCVRs(cvc, cvObj, svcObj, volumePolicy)
```

Comment:
isn't the check for `cvc.Publish.NodeID` being NULL need to be removed in caller of createVolumeOperation?

Reply:
`NodeID` will not be null, we will be having the picked one of the preferred node name everytime from the driver side as, it will available during volume create request at csi driver side.

Reply:
how can CSI driver fill the NodeID when there is no `allowedTopologies` or `WaitForFirstConsumer` is set in storageclass?